contracts/Marketplace.sol

```diff
@@ -324,6 +327,10 @@ contract Marketplace is Ownable, IMarketplace2 {
         if (requirePayment){
             price = getPriceInData(addSeconds, p.pricePerSecond, p.priceCurrency);
             require(datacoin.transferFrom(msg.sender, p.beneficiary, price), "error_paymentFailed");
+            if(price > 0 && txFee > 0){
+                uint256 fee = txFee.mul(price).div(1 ether);
+                require(datacoin.transferFrom(msg.sender, owner, fee), "error_paymentFailed");
```

Comment:
The fee should probably work like this:
- buyer pays a total of `price`
- beneficiary gets `price - fee`
- owner gets `fee`

Currently works like this:
- buyer pays a total of `price + fee` (fails because allowance is only granted for `price`)
- beneficiary gets `price`
- owner gets `fee`

Reply:
ok, but then the seller doesn't actually get the pricePerSecond they specified. I thought about this, but figured that modifying allowance to price + fee was best.

Reply:
Let's make it work the other way please. From the buyer's point of view the transaction fee is included in the price, and from the seller's point of view the seller pays the transaction fee. Yeah that's the way we want it.

Reply:
I updated the code to subtract fee from price. should onPurchase event show _price_ or _price - fee_?

Reply:
Event? You mean the "callback function" (for lack of a better name)? Could it take as argument both the net price (`price-fee`) as well as the `fee`? Any consequences of changing the API?

Reply:
Changes so far look good to me.

Reply:
Yeah I meant the PurchaseListener.onPurchase callback. I think we can change the signature to include price and fee, but I need to fix tests.

The TODO's re PurchaseListener involve decoding the returnData and requiring accepted = true if the contract does implement PurchaseListener. 

This call() syntax works even if the contract doesn't implement PurchaseListener:
 p.beneficiary.call(0x91517bdd, productId, subscriber, oldSub.endTimestamp, price);

but we need to unpack returnData